tic-tac-toe/src/index.js

```diff
@@ -0,0 +1,126 @@
+import React, { useState } from "react";
+import ReactDOM from "react-dom";
+import "./index.css";
+import * as serviceWorker from './serviceWorker';
+
+function Square({ value, onClick }) {
+
+  return (
+    <button className="square" onClick={onClick}>
+      {value}
+    </button>
+  );
+}
+
+ReactDOM.render(<Square />, document.getElementById("root"));
+
+function Restart({ onClick }) {
+
+  return (
+    <button className="restart" onClick={onClick}>
+      Play again
+    </button>
+  );
+}
+
+function Game() {
+	const [ squares, setSquares ] = useState(Array(9).fill(null));
+	const [ isXNext, setIsXNext ] = useState(true);
+	const nextSymbol = isXNext ? "X" : "O";
+	const winner = calculateWinner(squares);
+
+	function renderRestartButton() {
+	    return (
+	      <Restart
+	        onClick={() => {
+	          setSquares(Array(9).fill(null));
+	          setIsXNext(true);
+	        }}
+	      />
+	    );
+	  }
+
+	function renderSquare(i) {
+	    return (
+	      <Square
+	        value={squares[i]}
+	        onClick={() => {
+	          if (squares[i] != null || winner != null) {
+	            return;
+	          }
+	          const nextSquares = squares.slice();
+	          nextSquares[i] = (isXNext ? "X" : "O");
+	          setSquares(nextSquares);
+
+	          setIsXNext(!isXNext); // toggle turns
+	        }}
+	      />
+	    );
+	  }
+	function getStatus() {
+	    if (winner) {
+	      return "Winner: " + winner;
+	    } else if (isBoardFull(squares)) {
+	      return "Draw!";
+	    } else {
+	      return "Next player: " + (isXNext ? "X" : "O");
+	    }
+	  }
+
+    return (
+	    <div className="container">
+	      <div className="game">
+	        <div className="game-board">
+	          <div className="board-row">
+	            {renderSquare(0)}
+	            {renderSquare(1)}
+	            {renderSquare(2)}
+	          </div>
+	          <div className="board-row">
+	            {renderSquare(3)}
+	            {renderSquare(4)}
+	            {renderSquare(5)}
+	          </div>
+	          <div className="board-row">
+	            {renderSquare(6)}
+	            {renderSquare(7)}
+	            {renderSquare(8)}
```

Comment:
Separating out the logic of rendering a square helped reduce code repetition and makes this part of the code easier to reason about. 👍 To take this a step further, this could refactored by mapping through the `squares`. That would mean, however, that the styles for these rows would need updating.